agent/src/beerocks/slave/agent_ucc_listener.cpp

```diff
@@ -231,3 +233,23 @@ bool agent_ucc_listener::handle_dev_set_config(std::unordered_map<std::string, s
  * been received.
  */
 std::string agent_ucc_listener::get_selected_backhaul() { return m_selected_backhaul; }
+
+bool agent_ucc_listener::get_mac_by_ssid(const std::string &ruid, const std::string &ssid,
+                                         std::string &value)
+{
+    auto db    = AgentDB::get();
+    auto radio = db->get_radio_by_mac(tlvf::mac_from_string(ruid), AgentDB::eMacType::RADIO);
+    if (!radio) {
+        value = "ruid " + ruid + " not found";
+        return false;
+    }
+
+    for (const auto &bssid : radio->front.bssids) {
+        if (bssid.ssid == ssid) {
+            value = tlvf::mac_to_string(bssid.mac);
+            return true;
+        }
+    }
```

Comment:
use algorithms instead of raw loops:
```suggestion
auto found_it = std::find_if(
    radio->front.bssids.begin(), 
    radio->front.bssids.end(), 
    [&ssid](const decltype(bssids)::value_type &current_bssid){
           return current_bssid.ssid == ssid; 
});

if (found_it != front.bssids.end()) {
        value = tlvf::mac_to_string(ssid);
}
```
(please ignore compilation errors, I just wrote the idea. probably the type of the lambda parameter is incorrect and need to be fixed)